_posts/2019-04-10-imperatywnie-vs-funkcyjnie.md

```diff
@@ -0,0 +1,118 @@
+---
+layout:    post
+title:     "Programowanie imperatywne vs funkcyjne"
+date:      2019-04-10 9:00:00 +0100
+published: false
+author:    Łukasz Sroczyński
+tags:
+    - backend
+    - kotlin
+    
+---
+
+Programowanie funkcyjne w ostatnich latach stało się bardzo popularne. Umożliwia nam to pisanie kodu, który jest czystszy, bezpieczniejszy, a przede wszystkim łatwo testowalny. Oczywiście FP ma również wady z czego jedną z nich może być zbyt duże pójście w abstrakcję co powoduje, że kod staje się niezrozumiały. Omówimy sobie w takim razie jakie są różnice pomiędzy imperatywnym podejściem, a tym funkcyjnym. 
```

Comment:
> przede wszystkim łatwo testowalny

Że zacytuję klasyka "testowanie programu to bardzo efektywny sposób na pokazanie, że istnieją w nim błędy, ale beznadziejnie nieodpowiedni, by udowodnić, że ich w nim nie ma" :)